src/utils.c

```diff
@@ -223,6 +223,33 @@ ssize_t pwrite(int fd, const void *buf, size_t count, uint64_t offset)
 }
 #endif
 
+int allocatefile(int fd, off_t offset, off_t length)
```

Comment:
Need to add windows support for allocatefile, I have a test at https://gist.github.com/braydonf/059146ac00c8d878bbb1e226ac8aa934 we can use.